### ISSUE TITLE
Unnecessary warning "Redis failed to start, retrying now" in ray.init()

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#8777]
When I run `import ray; ray.init()`, I see the warning 

```
2020-06-03 23:51:08,991	WARNING services.py:923 -- Redis failed to start, retrying now.
```

which is not useful. Redis is starting up in the background (asynchronously), and this just indicates that Redis hasn't started up yet.

Some notes about the fix:
- We _do_ want to see this error message if/when Redis actually fails to start after a reasonable amount of time (e.g., 5 seconds).
- We don't want to sleep for long between retries because that will slow down `ray.init()`.

https://github.com/ray-project/ray/pull/8475 addressed a related (but not the same) issue. Maybe there are more?

**Comment by rightx2** [comment]
I'm the newbie of ray and I thought this was an error.... Have spent so much time in trying fix it :(

**Comment by rkooo567** [comment]
@rightx2 oh.. I am so sorry about that. Could you tell us what we could’ve done differently to help you avoiding this kind of issue again? It’ll be really helpful!

**Comment by robertnishihara** [comment]
@rkooo567 I suppose we could have gotten rid of this error message earlier :)

**Comment by Hdong179** [comment]
I encountered the same problem in the process of learning, 
`2020-07-28 10:42:23,295 WARNING services.py:923 -- Redis failed to start, retrying now.
2020-07-28 10:42:23,545 INFO services.py:1165 -- View the Ray dashboard at localhost:8265
[0, 1, 4, 9]`
I found that the python version may not match the downloaded ray。
After confirming that I am python3.7, 
`pip install https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.9.0.dev0-cp37-cp37m-manylinux1_x86_64.whl`
I re-downloaded and solved the problem

**Comment by seelam** [comment]
I got this message... it is very confusing for the following reasons:
1. It gives a `WARNING that Redis failed to start, retrying now`, 

If it had connected, I would expect a message that it connected. It should print that it did connect so please add something to that effect.

2. The last line it prints is `ray stop`, which I thought the cluster stopped. Best to remove the space between the last statement ` have been started, run` and `ray stop`.



```
ubuntu@seelam-ansible-host:~$ ray start --head --port=6379
2020-10-15 00:42:52,764	INFO scripts.py:395 -- Using IP address 10.242.128.4 for this node.
2020-10-15 00:42:52,893	INFO resource_spec.py:212 -- Starting Ray with 12.55 GiB memory available for workers and up to 6.3 GiB for objects. You can adjust these settings with ray.init(memory=<bytes>, object_store_memory=<bytes>).
2020-10-15 00:42:53,104	WARNING services.py:923 -- Redis failed to start, retrying now.
2020-10-15 00:42:53,288	INFO services.py:1165 -- View the Ray dashboard at localhost:8265
2020-10-15 00:42:53,305	INFO scripts.py:425 -- 
Started Ray on this node. You can add additional nodes to the cluster by calling

    ray start --address='10.242.128.4:6379' --redis-password='5241590000000000'

from the node you wish to add. You can connect a driver to the cluster from Python by running

    import ray
    ray.init(address='auto', redis_password='5241590000000000')

If you have trouble connecting from a different machine, check that your firewall is configured properly. If you wish to terminate the processes that have been started, run

    ray stop
```
